Neos.ContentRepository.Core/Classes/Projection/Workspace/Workspace.php

```diff
@@ -24,31 +24,72 @@
  *
  * @api
  */
-class Workspace
+final readonly class Workspace
 {
     /**
      * This prefix determines if a given workspace (name) is a user workspace.
      */
     public const PERSONAL_WORKSPACE_PREFIX = 'user-';
 
+    /**
+     * @var WorkspaceName Workspace identifier, unique within one Content Repository instance
+     */
+    public WorkspaceName $workspaceName;
+
+    /**
+     * @var WorkspaceName|null Workspace identifier of the base workspace (i.e. the target when publishing changes) – if null this instance is considered a root (aka public) workspace
+     */
+    public ?WorkspaceName $baseWorkspaceName;
+
+    /**
+     * @deprecated with 9.0.0-beta12 metadata should be assigned to workspaces outside the Content Repository core
+     */
+    public WorkspaceTitle $workspaceTitle;
+
+    /**
+     * @deprecated with 9.0.0-beta12metadata should be assigned to workspaces outside the Content Repository core
+     */
+    public WorkspaceDescription $workspaceDescription;
+
+    /**
+     * The Content Stream this workspace currently points to – usually it is set to a new, empty content stream after publishing/rebasing the workspace
+     */
+    public ContentStreamId $currentContentStreamId;
+
+    /**
+     * The current status of this workspace
+     */
+    public WorkspaceStatus $status;
+
+    /**
+     * @deprecated with 9.0.0-beta12 owners/collaborators should be assigned to workspaces outside the Content Repository core
+     */
+    public string|null $workspaceOwner;
```

Comment:
did that even work before ... and how does the deprecation strategy look like? Can we just not remove it with the migration in place?

Reply:
Also `CreateWorkspace:$workspaceOwner` is not deprecated?

Reply:
> Can we just not remove it with the migration in place?

What do you mean?

> Also CreateWorkspace:$workspaceOwner is not deprecated?

Right, that has to be deprecated. On it